editorconfig/editorconfig#429 Remove section name, key and value length limits (#21)
editorconfig/specification: index.rst

The backslash character (``\\``) can be used to escape a character so it is
not interpreted as a special character.

> The minimum supported section name length is 1024 characters.

Reviewer: To a new reader, this may sound like each key has to be at least 1024 chars long. I wonder if something like the following would be better 

```suggestion
Cores must accept section names with length up to 1024 characters.
```

Author: Tre tricky part here is that we have to write for two types of readers: editor config users and core/plugin devs. .)

Your suggestion hits the swet spots. I'd say. Thank you once more.

> Cores must accept section names with length up to 1024 characters.

Reviewer 2: Request: "up to and including 1024".  That reduces the chance of an off-by-one error.

Author: good catch! Fixed. Thanks. .)

> Beyond that, each implementation can define its own upper limit.

Reviewer: ```suggestion
Beyond that, each implementation may choose to define its own upper limit or no explicit limit at all.
```

Author: Oh yeah thanks a lot for this! <3 (and the other suggestion)

unset`` to undefine the ``indent_size`` pair (and use editor defaults).

Pair keys are case insensitive. All keys are lowercased after parsing.

> The minimum supported length of key and value is 1024 and 4096 characters respectively.

Reviewer: Similar like above:

```suggestion
Cores must accept section keys and values with lengths up to 1024 and 4096 characters respectively.
```

> Cores must accept keys and values with lengths up to 1024 and 4096 characters respectively.

Reviewer 2: Same - "up to and including"

Author: fixed too

> Beyond that, each implementation can define its own upper limit.

Reviewer: ```suggestion
Beyond that, each implementation may choose to define its own upper limits or no explicit limits at all.
```
